lib/revert-global.js

```diff
@@ -0,0 +1,47 @@
+// Track global and require.cache keys and
+// removes any new keys when revert is called
+function revertGlobal() {
+	const storedKeys = {};
+	const storedCache = Object.assign({}, require.cache);
+	const storedExtensions = Object.assign({}, module.constructor._extensions);
+	const preventCache = ['bluebird', '../index.js', './worker/main.js'];
```

Comment:
Why is Bluebird being prevented from caching? 

I've tried running my tests against this branch, and all of our tests that involve bluebird cancellation are failing: we currently have a 'testBootstrap.js" file the we import before all of our tests which enables bluebird cancellation, but it's not being respected, likely due to this logic here.

Reply:
Perhaps `.getNewLibraryCopy()` would be useful so that the test framework can have its own copy of Bluebird without stepping on the application code?

Reply:
I was clearing bluebird there to try to not interfere with the test's usage of bluebird but you might be right that clearing it there ended up interfering anyways. I moved it to only clear initially so hopefully it doesn't conflict now.

Reply:
Yup, it looks like that fixed it.  All of my cancellation tests are running correctly now.